compiler/src/iree/compiler/Codegen/LLVMGPU/test/reduction_pipeline_softmax_rocm.mlir

```diff
@@ -95,8 +95,6 @@ func.func @dynamic_softmax() {
   return
 }
 
-
-// Finer details of this lowering are captured by the spirv pipeline test. Just
-// verify that warp reduction triggers.
+// Verify that LLVMVectorDistribute is used for dynamic softmax.
+// CHECK: LLVMGPUVectorDistribute
 //    CHECK-LABEL: func.func @dynamic_softmax
```

Comment:
How does the reduction look like now?

Reply:
I think I understand what is happening here... This PR sets the the reduction size to target (for dynamic reduction sizes) to be very large (1<<31). Because of this, an entire workgroup is allocated to a single row (each workgroup reduces 1<<31 values to 1 value). The subgroup_reduce above is doing this full reduce. So it looks correct to me. 


i.e. subgroup_reduce always reduces across all subgroups, and a subset of threads. But the subset of threads is 32 (size = 32) -- i.e. all threads. So all 32 threads and all 32 subgroups.

Reply:
Please confirm that this seems reasonable and this is good to land @kuhar